File: acm-java/src/main/java/leetcode/LeetCode2433.java
package leetcode;

public class LeetCode2433 {
    public int[] findArray(int[] pref) {
        var ans = new int[pref.length];
        for (int i = 0; i < pref.length; i++) {
            ans[i] = (i == 0 ? 0 : pref[i - 1]) ^ pref[i];
        }
        return ans;
    }
}

File: acm-kotlin/src/main/kotlin/leetcode/LeetCode2433.kt
package leetcode

class LeetCode2433 {
    fun findArray(pref: IntArray): IntArray {
        return (0..< pref.count())
            .map { i ->
                (if (i > 0) pref[i - 1] else 0) xor pref[i]
            }
            .toIntArray()
    }
}